boost.sh

```diff
@@ -1404,7 +1417,13 @@ EXTRA_IOS_FLAGS="$EXTRA_FLAGS -fembed-bitcode -mios-version-min=$MIN_IOS_VERSION
 EXTRA_TVOS_FLAGS="$EXTRA_FLAGS -fembed-bitcode -mtvos-version-min=$MIN_TVOS_VERSION"
 EXTRA_OSX_FLAGS="$EXTRA_FLAGS -mmacosx-version-min=$MIN_OSX_VERSION"
 EXTRA_ANDROID_FLAGS="$EXTRA_FLAGS"
-EXTRA_LINUX_FLAGS="$EXTRA_FLAGS"
+
+if [[ -n "$USE_CXX11_ABI" ]]; then
+   EXTRA_LINUX_FLAGS="$EXTRA_FLAGS -D_GLIBCXX_USE_CXX11_ABI=0"
```

Comment:
From [docs](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html):

> Using the default configuration options for GCC the default value of the macro is 1 which causes the new ABI to be active, so to use the old ABI you must explicitly define the macro to 0 before including any library headers. (`Be aware that some GNU/Linux distributions configure GCC 5 differently so that the default value of the macro is 0 and users must define it to 1 to enable the new ABI.`)

This means that you should set this flag explicitly in all cases, not assuming the default.

The proper support matrix must be:
USE_CXX11_ABI==empty: non-ABI specifying build, no flag
USE_CXX11_ABI==1: -D_GLIBCXX_USE_CXX11_ABI=1
USE_CXX11_ABI==0: -D_GLIBCXX_USE_CXX11_ABI=0

Reply:
> This means that you should set this flag explicitly in all cases, not assuming the default.

Are you saying that you want regular linux builds to explicitly set `-D_GLIBCXX_USE_CXX11_ABI=1`?

> The proper support matrix must be:

Yes, I understand your previous comment about `USE_CXX11_ABI` being inverted. I can fix this and add support for all 3 cases, but is anyone going to define `USE_CXX11_ABI==1`?